lib/ably/realtime/presence/presence_manager.rb

```diff
@@ -19,13 +19,18 @@ def initialize(presence)
         setup_channel_event_handlers
       end
 
-      # Expect SYNC ProtocolMessages from the server with a list of current members on this channel
-      #
-      # @return [void]
-      #
       # @api private
-      def sync_expected
-        presence.members.change_state :sync_starting
+      def on_attach(has_presence_flag)
+        if has_presence_flag
+          # Expect SYNC ProtocolMessages from the server with a list of current members on this channel
+          presence.members.change_state :sync_starting
+        else
+          # There server has indicated that there are no SYNC ProtocolMessages to come because
+          # there are no members on this channel
+          logger.debug { "#{self.class.name}: Emitting leave events for all members as a SYNC is not expected and thus there are no members on the channel" }
+          presence.members.change_state :sync_none
+        end
+        presence.members.send(:enter_local_members)
```

Comment:
Why not `presence.members.enter_local_members`?

Reply:
Actually, `enter_local_members` is a private method. Since, we don't want to expose it as a public method, we have marked it as private. If try to call private method directly, ruby throws runtime error.

Reply:
We really shouldn’t be having one class calling a private method of another class; that doesn't seem right. I mean, in this case `enter_local_members` is a private method on `MembersMap`, which is a class that isn't even part of the public API of the library, right? So I think it'd be fine for `enter_local_members` to be public?

Reply:
Fixed https://github.com/ably/ably-ruby/pull/410/commits/446a16fabb036ff135b765f0bcce5a88bf09b11e